### PR TITLE
Issue #2186: Exception message extended to provide more information on incorrect parent

### PR DESCRIPTION
**Pull request by subkrish**
Issue #2186

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java [hunk 1]
```diff
@@ -152,7 +152,9 @@ public void setupChild(Configuration childConf)
         final Object module = moduleFactory.createModule(name);
         if (!(module instanceof AbstractCheck)) {
             throw new CheckstyleException(
-                "TreeWalker is not allowed as a parent of " + name);
+                "TreeWalker is not allowed as a parent of " + name
+                        + " Please review 'Parent Module' section for this Check in web"
+                        + " documentation if Checks is standard.");
```
Comment (Vladlis) [inline src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java]: Checks -> Check

Comment (subkrish) [inline src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java]: Done.

**Comment by rnveach** [comment]
No tests need changing because the current test just checks the start of the message and we only changed the end.
https://github.com/checkstyle/checkstyle/blob/7c2f448ad61f66224168d197dfc6195e7b8d9409/src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java#L130

**Comment by rnveach** [comment]
Since we are changing this message, I think we should also change the default one in `AutomaticBean`.
https://github.com/checkstyle/checkstyle/blob/3bd3a52280388520c64ec813dd1e4ffb3a0063a2/src/main/java/com/puppycrawl/tools/checkstyle/api/AutomaticBean.java#L273
> childConf.getName() + " is not allowed as a child in " + getConfiguration().getName()

@romani You ok with this?

**Comment by romani** [comment]
@subkrish , please apply suggestion from @rnveach . I am ok with new message. Ready to merge.

**Comment by subkrish** [comment]
@romani @rnveach Changes have been made. Please review.

**Comment by rnveach** [comment]
@romani Please review and merge.

**Pull request merged**